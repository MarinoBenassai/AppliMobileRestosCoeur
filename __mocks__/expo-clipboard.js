export function Clipboard() {

    const clip = {setString: setString};
 
    return clip;
}

export const setString = jest.fn();