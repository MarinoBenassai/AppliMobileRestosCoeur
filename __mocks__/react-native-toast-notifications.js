export function useToast() {

    const toast = {show: show};
 
    return toast;
}

export const show = jest.fn();