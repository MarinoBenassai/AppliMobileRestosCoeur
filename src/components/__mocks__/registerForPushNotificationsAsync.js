import * as Device from 'expo-device';
import * as Notifications from 'expo-notifications';

// Fonction de cr√©ation/registration du token de notification
export const  registerForPushNotificationsAsync = async function (device) {
	let token = "-1";
	if (Device.isDevice) {
		if (!Device.brand){//if(device != "1" && device != "2"){
			return token;
		}
		token = "ExponentPushToken[999999999]";

	} else {
	  	alert('Must use physical device for Push Notifications');
	}
  
	return token;
}