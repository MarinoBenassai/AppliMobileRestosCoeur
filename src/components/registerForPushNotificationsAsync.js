import * as Device from 'expo-device';
import * as Notifications from 'expo-notifications';
import React, { useEffect, useState } from 'react';
import { StyleSheet, Button, Text, View, Image, TextInput, Pressable, ActivityIndicator} from 'react-native';

// Fonction de création/registration du token de notification
export const  registerForPushNotificationsAsync = async function (device) {
	let token = "-1";
	if (Device.isDevice) {
		if(device != "1" && device != "2"){
			console.log("on sort");
			return token;
		}
		const { status: existingStatus } = await Notifications.getPermissionsAsync();
		let finalStatus = existingStatus;
		if (existingStatus !== 'granted') {
			const { status } = await Notifications.requestPermissionsAsync();
			finalStatus = status;
		}

		// Notification non autorisée
		if (finalStatus !== 'granted') {
			alert('Failed to get push token for push notification!');
			return token;
		}

		token = (await Notifications.getExpoPushTokenAsync()).data;
		console.log(token);

	} else {
	  	alert('Must use physical device for Push Notifications');
	}
  
	if (Platform.OS === 'android') {
		Notifications.setNotificationChannelAsync('default', {
			name: 'default',
			importance: Notifications.AndroidImportance.MAX,
			vibrationPattern: [0, 250, 250, 250],
			lightColor: '#FF231F7C',
		});
	}
  
	return token;
}