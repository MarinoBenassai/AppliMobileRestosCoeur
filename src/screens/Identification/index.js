import React, { useEffect, useState } from 'react';
import { StyleSheet, Button, Text, View, Image, TextInput, Pressable, ActivityIndicator} from 'react-native';
import * as Crypto from 'expo-crypto';
import * as Notifications from 'expo-notifications';
import Constants from 'expo-constants';

import constantes from '../../constantes';
import {userContext} from '../../contexts/userContext';
import styles from '../../styles';
import logo from '../../../assets/logoRdC.png'


export default function IdScreen({navigation}) {

	const [textEmail, onChangeTextEmail] = React.useState('');
	const [textPassword, onChangeTextPassword] = React.useState('');

	const [loading, setLoading] = React.useState(false);

	const [expoPushToken, setExpoPushToken] = useState('');
	
	const changeID = React.useContext(userContext).changeID;
	const changeToken = React.useContext(userContext).changeToken;
    const handleError = React.useContext(userContext).handleError;

	// On va chercher le token de notification
/* 	useEffect(() => {
		registerForPushNotificationsAsync().then(token => setExpoPushToken(token));
	}, []); */

  
  function checkPassword() {
	if (textEmail != '' && textPassword != '') {
		setLoading(true);
		fetch('http://' + constantes.BDD + '/Axoptim.php/AUT/AP_LOGIN/P_EMAIL=' + textEmail +"/P_MOTDEPASSE=" + textPassword)

			.then((response) => {
				if (response.ok) {
					const a = response.json();
					const b = registerForPushNotificationsAsync();

					return Promise.all([a, b])
				}
				else {
					const json = response.json();
					throw new Error(json['error']);
				}
			})
			
			
			.then(([data, token]) => {
				fetch('http://' + constantes.BDD + '/Axoptim.php/REQ/AP_UPD_NOTIF/P_IDBENEVOLE=' + data.id + '/P_TOKENNOTIF=' + token + "/P_TOKEN=" + data.token);
				login(data);
			})

			//.then((response) => response.text())
			//.then(texte => console.log(texte))
			.catch((error) => {handleError(error); setLoading(false)})
		  

		onChangeTextPassword(''); //On vide le champ mot de passe
	}
  }
  
  
  function login(data) {
	  changeToken(data.token);
	  changeID(data.id);
    }
  
  return (
    <>
    <View style={[styles.container,{alignItems: "center"}]}>
	  <Image
		style={styles.logo}
        source={logo}
      />
      <Text style={styles.idTexte}>Adresse mail</Text>
	  <TextInput
		style={styles.idInput}
		placeholder="votre@email.fr"
		onChangeText={onChangeTextEmail}
		keyboardType = "email-address"
		textContentType = "emailAddress"
		autoCompleteType = "email"
		autoCorrect = {false}
	  />
	  <Text style={styles.idTexte}>Mot de passe</Text>
	  <TextInput
		style={styles.idInput}
		placeholder="********"
		secureTextEntry = {true}
        onChangeText={onChangeTextPassword}
		value={textPassword}
		textContentType = "password"

	  />
	  <Button
		onPress = {() => checkPassword()} //{() => navigation.navigate('Engagements')}
		title="CONNEXION"
		color="#e7007d"
		accessibilityLabel="Bouton de connexion"

	  />
	  <Pressable title = "mdp" onPress = {() => navigation.navigate("Oublie")} >
		<Text style={styles.idTexte}>Mot de passe oublié ?</Text>
	  </Pressable>
	  	
    <Button
	  onPress = {() => {changeID(constantes.IDDebug); changeToken(constantes.IDDebug)}}
	  title="PASSER"
	  color="#041584"
    />
    </View>
	{loading &&
    <View style={styles.loading}>
      <ActivityIndicator size="large" color="#00ff00" />
    </View>
	}
    </>
  );
}


// Fonction de création/registration du token de notification
async function registerForPushNotificationsAsync() {
	let token;
	if (Constants.isDevice) {
		const { status: existingStatus } = await Notifications.getPermissionsAsync();
		let finalStatus = existingStatus;
		if (existingStatus !== 'granted') {
			const { status } = await Notifications.requestPermissionsAsync();
			finalStatus = status;
		}

		// Notification non autorisée
		if (finalStatus !== 'granted') {
			alert('Failed to get push token for push notification!');
			//return;
		}

		token = (await Notifications.getExpoPushTokenAsync()).data;
		console.log(token);

	} else {
	  	alert('Must use physical device for Push Notifications');
	}
  
	if (Platform.OS === 'android') {
		Notifications.setNotificationChannelAsync('default', {
			name: 'default',
			importance: Notifications.AndroidImportance.MAX,
			vibrationPattern: [0, 250, 250, 250],
			lightColor: '#FF231F7C',
		});
	}
  
	return token;
}
