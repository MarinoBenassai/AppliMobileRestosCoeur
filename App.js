import React from 'react';
import { useState, useEffect, useRef} from 'react';
import { useWindowDimensions } from 'react-native';
import {ActivityIndicator, StyleSheet, Button, Text, View, Image, TextInput, PixelRatio} from 'react-native';
import Ionicons from '@expo/vector-icons/Ionicons';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { createDrawerNavigator } from '@react-navigation/drawer';
import Icon from 'react-native-vector-icons/Octicons';
import Constants from 'expo-constants';
import * as SecureStore from 'expo-secure-store';
import * as Device from 'expo-device';
import * as SplashScreen from 'expo-splash-screen';
import * as Notifications from 'expo-notifications';

import IdScreen from './src/screens/Identification';
import compteScreen from './src/screens/Compte';
import contactScreen from './src/screens/Contacts';
import engagementScreen from './src/screens/Engagements';
import referentScreen from './src/screens/SynthRef';
import activiteScreen from './src/screens/Activite';
import oublieScreen from './src/screens/MdpOublie';
import informationScreen from './src/screens/Informations';

import listeUtilisateurScreen from './src/screens/ListeUtilisateur';
import {checkFetch} from './src/components/checkFetch';
import {userContext} from './src/contexts/userContext';

import constantes from './src/constantes';

import {registerForPushNotificationsAsync} from "./src/components/registerForPushNotificationsAsync.js";

import { ToastProvider } from 'react-native-toast-notifications'

const engagementStack = createStackNavigator();
const synthRefStack = createStackNavigator();
const compteStack = createStackNavigator();
const contactStack = createStackNavigator();
const identificationStack = createStackNavigator();
const informationStack = createStackNavigator();

const Drawer = createDrawerNavigator();

// handler de notification
Notifications.setNotificationHandler({
  handleNotification: async () => ({
    shouldShowAlert: true,
    shouldPlaySound: false,
    shouldSetBadge: false,
  }),
});

export default function App() {
	
  const [userID,setUserID] = React.useState("");
  const [token, setToken] = React.useState("");
  const [estReferent, setReferent] = React.useState(true);
  const windowWidth = useWindowDimensions().width;
  const windowHeight = useWindowDimensions().height;

  //Bulle d'info
  const fSize = 12;
  const [X,setX] = React.useState(0);
  const [Y,setY] = React.useState(0);
  const [bulleVisible, setBulleVisible] = React.useState(false);
  const [messageBulle, setMessageBulle] = React.useState("");
  const [horizPos,setHorizPos] = React.useState('left');
  const [vertPos,setVertPos] = React.useState('up');
  
  function afficherInfoBulle (pageX, pageY, message){
	console.log(windowWidth,windowHeight,pageX, pageY, PixelRatio.get());//, getPixelSizeForLayoutSize(windowWidth), getPixelSizeForLayoutSize(windowHeight));
	getCoordFromText(message, pageX, pageY);
	setBulleVisible(true);
  }
  
  function cacherInfoBulle (){
	setBulleVisible(false);
  }

  function getCoordFromText(comment, X, Y) {
	setY(Y);
    setX(X);
	//On calcule les dimensions maximum de la bulle d'aide
	var horizSpace = 0;
	var vertSpace = 0;

	if (X > windowWidth/2){
		//Il y a plus d'espace à gauche
		setHorizPos('left');
		horizSpace = X;
	}
	else {
		//Il y a plus d'espace à droite
		setHorizPos('right');
		horizSpace = windowWidth - X;
	}
	
	if (Y > windowHeight/2){
		//Il y a plus d'espace vers le haut
		setVertPos('up');
		vertSpace = Y;
	}
	else {
		//Il y a plus d'espace vers le bas
		setVertPos('down');
		vertSpace = windowHeight - Y;
	}

	var formattedComment = [];
	const maxNbLines = Math.floor(vertSpace/(fSize * PixelRatio.get()));
	const maxLength = Math.floor(horizSpace/(fSize * PixelRatio.get()));
	var currentLength = 0;
	var nbLines = 0;
	
	const lines = comment.split("\n");
	for (let i = 0; i<lines.length; i++) {
		nbLines += 1;
		const words = lines[i].split(" ");
		for (let j = 0; j<words.length; j++){
			if ((currentLength + words[j].length) <	maxLength) {
				currentLength += words[j].length
				formattedComment.push(words[j]);
			}
			else {
				if ((nbLines + 1) >maxNbLines){
					formattedComment.push("...")
					setMessageBulle(" " + formattedComment.join(" ").trim());
					return;
				}
				else {
					nbLines += 1;
					currentLength = 0;
					formattedComment.push(words[j] + "\n");
				}
			}
		}
		if ((nbLines + 1) > maxNbLines){
			formattedComment.push("...")
			setMessageBulle(" " + formattedComment.join(" ").trim());
			return;
		}
		else {
			nbLines += 1;
			currentLength = 0;
			formattedComment[formattedComment.length - 1] += "\n";
		}
	}	
	setMessageBulle(" " + formattedComment.join(" ").trim());
  }

  const styles = StyleSheet.create({
  bulle: {
    backgroundColor: '#FFFFFFFF',
    alignItems: 'center',
    justifyContent: 'center',
	padding: 5,
	borderColor: 'black',
	borderWidth: 1,
  },
  bullePlacer: {
    position: 'fixed',
	width: horizPos === 'left' ? X : 'auto',
    left: horizPos === 'left' ? 0 : X,
	height: vertPos === 'up' ? Y : 'auto',
    top: vertPos === 'up' ? 0 : Y,
    backgroundColor: 'transparent',
	flexDirection: 'row',
	justifyContent: horizPos === 'left' ? 'flex-end' : 'flex-start',
	alignItems: vertPos === 'up' ? 'flex-end' : 'flex-start'
  },
  });

  // Notification
  const [notification, setNotification] = useState(false);
  const notificationListener = useRef();
  const responseListener = useRef();

  useEffect(() => {
    notificationListener.current = Notifications.addNotificationReceivedListener(notification => {
      setNotification(notification);
    });

    responseListener.current = Notifications.addNotificationResponseReceivedListener(response => {
      console.log(response);
    });

    return () => {
      Notifications.removeNotificationSubscription(notificationListener.current);
      Notifications.removeNotificationSubscription(responseListener.current);
    };
  }, []);
  
  useEffect(() => {
	  autoConnect()
	  .catch((error) => handleError(error));
  }, []);
  
  async function sendAPI(apCode,sqlCode,params) {
	  let body = new FormData();
	  body.append('params',JSON.stringify(params));
	  if (apCode === 'APP') {
	    body.append('token',token);
	  }
	  const response = await fetch('http://' + constantes.BDD + '/' + apCode + '/' + sqlCode + '/', {
	    method: 'POST',
	    body: body});
	  return checkFetch(response,apCode);
  }
  
  async function autoConnect() {
	  
	  if (Device.brand){
		const device = await Device.getDeviceTypeAsync();
		await SplashScreen.preventAutoHideAsync();
		const token = await SecureStore.getItemAsync('token');
		const id = await SecureStore.getItemAsync('id');
		if (token !== null && id !== null){
			setToken(token);
			setUserID(id);
			const tokennotif = await registerForPushNotificationsAsync(device);
			sendAPI('APP', 'AP_UPD_NOTIF', {'P_IDBENEVOLE':id, 'P_TOKENNOTIF':tokennotif})
			.catch((error) => handleError (error));
		}
		await SplashScreen.hideAsync();
	  }
	  
  }
  
  function changeID(ID) {
    setUserID(ID);
  }
  
  function changeToken(token) {
    setToken(token);
  }
  
  async function logout() {
	const device = await Device.getDeviceTypeAsync();
	const tokennotif = await registerForPushNotificationsAsync(device);
	
	sendAPI('APP','AP_LOGOUT',{'P_TOKEN':token,'P_TOKENNOTIF':tokennotif})
	.then((data) => {
		setUserID("");
		setToken("");
		if (Device.brand){
			SecureStore.deleteItemAsync('id');
			SecureStore.deleteItemAsync('token');
		}
	})
	.catch((error) => handleError(error));
  }
  
  function handleError (erreur) {
	  if (erreur === "Invalid token"){
		alert("Votre session a expiré, veuillez vous reconnecter");
		setUserID("");
		setToken("");
		if (Device.brand){
			SecureStore.deleteItemAsync('id');
			SecureStore.deleteItemAsync('token');
		}
		logout();
	  }
	  else {
		alert(erreur);
	  }
  }
  
  return (
	<ToastProvider>
		<userContext.Provider value = {{
		userID: userID,
		token: token,
		logoutUser: logout,
		changeID: changeID,
		changeToken: changeToken,
		handleError: handleError,
		sendAPI: sendAPI,
		setReferent: setReferent,
		afficherInfoBulle: afficherInfoBulle,
		cacherInfoBulle: cacherInfoBulle
		}}>
			<NavigationContainer>
				<Drawer.Navigator screenOptions = {{swipeEnabled : false}}>
					{userID === "" ? (
						<Drawer.Screen name="Identification" component = {identification}/>		
					) : (
						<>
						<Drawer.Screen name="Engagements" component={engagement} options={{ title: "Mes engagements" }} />
						{estReferent && <Drawer.Screen name="SynthRef" component={referent} options={{ title: "Ma synthèse référent" }} />}
						<Drawer.Screen name="Compte" component={compte} options={{ title: "Mon compte" }} />
						<Drawer.Screen name="Contacts" component={contact} options={{ title: "Mes contacts" }} />
						<Drawer.Screen name="Informations" component={information} options={{ title: "Mes informations" }} />
						</>
					)}
				</Drawer.Navigator>
			</NavigationContainer>
			{(bulleVisible &&
			<View pointerEvents="none" style={styles.bullePlacer}>
				<View pointerEvents="none" style={styles.bulle}>
					<Text style = {{fontSize: fSize}}>{messageBulle}</Text>
				</View>
			</View>)}
		</userContext.Provider>
	</ToastProvider>
  );
  
}

const boutonMenu = ({nav}) => <Icon 
    name='three-bars' 
    size={30}
    color='#000' 
    onPress={nav.openDrawer}
/>;

const boutonLogOut = () => <Icon 
              onPress={React.useContext(userContext).logoutUser}
			  name='sign-out' 
			  size={30}
			  color='#000'
            />;


function identification() {
  return (
    <identificationStack.Navigator>
	  <identificationStack.Screen name="Identification" component = {IdScreen} options = {{title: "Se connecter"}}/>
	  <identificationStack.Screen name="Oublie" component={oublieScreen} options={{ title: "Mot de passe oublié" }} />
    </identificationStack.Navigator>
  );
}

const screenOptionsBase = {headerRight: () => (boutonLogOut()), headerStyle: {backgroundColor: '#41b495',}, headerRightContainerStyle: {paddingRight: 20},};
function screenOptionsFirstPage (nav, title) {return {title: title, headerLeft: () => (boutonMenu({nav})), headerLeftContainerStyle: {paddingLeft: 20},}}

function engagement({navigation}) {
  const nav = navigation;
  return (
    <engagementStack.Navigator screenOptions = {screenOptionsBase}>
	  <engagementStack.Screen name="Engagements" component={engagementScreen} options={screenOptionsFirstPage(nav, "Mes engagements")}/>
	  <engagementStack.Screen name="Activite" component={activiteScreen} options={{ title: "Détails" }} />
      <engagementStack.Screen name="ListeUtilisateur" component={listeUtilisateurScreen} options={{title: "Ajout d’un bénévole"}}/>
    </engagementStack.Navigator>
  );
}

function referent({navigation}) {
  const nav = navigation;
  return (
    <synthRefStack.Navigator screenOptions= {screenOptionsBase}>
	  <synthRefStack.Screen name="SynthRef" component={referentScreen} options={screenOptionsFirstPage(nav, "Ma synthèse référent") } />
	  <synthRefStack.Screen name="Activite" component={activiteScreen} options={{ title: "Détails" }} />
	  <synthRefStack.Screen name="ListeUtilisateur" component={listeUtilisateurScreen} options={{title: "Ajout d’un bénévole"}}/>
	</synthRefStack.Navigator>
  );
}

function compte({navigation}) {
  const nav = navigation;
  return (
    <compteStack.Navigator screenOptions={screenOptionsBase}>
	  <compteStack.Screen name="Compte" component={compteScreen} options={screenOptionsFirstPage(nav,"Mon compte")} />
    </compteStack.Navigator>
  );
}

function contact({navigation}) {
  const nav = navigation;
  return (
    <contactStack.Navigator screenOptions={screenOptionsBase}>
	  <contactStack.Screen name="Contacts" component={contactScreen} options={screenOptionsFirstPage(nav, "Mes contacts")}/>
    </contactStack.Navigator>
  );
}

function information({navigation}) {
	const nav = navigation;
	return (
	  <informationStack.Navigator screenOptions={screenOptionsBase}>
		<informationStack.Screen name="Informations" component={informationScreen} options={screenOptionsFirstPage(nav, "Mes Informations")}/>
	  </informationStack.Navigator>
	);
  }

