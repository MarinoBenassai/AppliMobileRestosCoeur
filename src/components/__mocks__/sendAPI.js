import constantes from '../../constantes';
import {checkFetch} from '../../components/checkFetch';

export const sendAPI =   async (apCode,sqlCode,params, tokenCo = null) => {
  var body = {};
  if (apCode === 'APP') {
	body = JSON.stringify({
	  'params' : params,
	  'token' : tokenCo,
	});
  }
  else {
	body = JSON.stringify({
	  'params' : params,
	});
  }
  const response = await fetch(constantes.ADDRESS +  '/' + apCode + '/' + sqlCode + '/', {
	method: 'POST',
	body: body});
  return checkFetch(response,apCode);
};