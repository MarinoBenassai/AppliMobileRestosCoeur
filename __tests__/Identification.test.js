import React from 'react';
import { cleanup, render, fireEvent, waitFor } from "@testing-library/react-native";

import Identification from '../src/screens/Identification';
import App from "../App.js"
import {userContext} from '../src/contexts/userContext'
import {registerForPushNotificationsAsync} from '../src/components/__mocks__/registerForPushNotificationsAsync'
jest.mock("../src/components/sendAPI.js");

afterEach(cleanup)

describe('Identification', () => {
	
  it('should display "CONNEXION" button', () => {

    const {getByText} = render(<Identification />);

    expect(getByText('CONNEXION').props.children).toEqual('CONNEXION');
  });
  
  it('should not display "PASSER" button', () => {

    const {queryByText} = render(<Identification />);

    const foundButton = queryByText("PASSER");

    expect(foundButton).toBeNull();
  });
  
  it('should fail on incorrect password', async () => {
	const handleError = jest.fn(() => {});	
	context = {
		handleError: handleError,
	}
	
    const { getByTestId, getByText, getByPlaceholderText } = render(
	<userContext.Provider value = {context}>
		<Identification />
	</userContext.Provider>
	);

	fireEvent.changeText(getByPlaceholderText('votre@email.fr'), "a@a.a");
	fireEvent.changeText(getByPlaceholderText('********'), "zzz");
    fireEvent.press(getByText('CONNEXION'));
	await waitFor(() => expect(handleError).toHaveBeenCalledTimes(1));
	expect(handleError).toBeCalledWith("Login ou mot de passe incorrect");
	  
  });
  
  it('should store id/token on correct login', async () => {

	const handleError = jest.fn(() => {});	
	const changeID = jest.fn(() => {});
	const changeToken = jest.fn(() => {});
	context = {
		changeID: changeID,
		changeToken: changeToken,
		registerForPushNotificationsAsync: registerForPushNotificationsAsync,
		handleError: handleError,
	}
	
	const { getByTestId, getByText, getByPlaceholderText } = render(
	<userContext.Provider value = {context}>
		<Identification />
	</userContext.Provider>
	);
	
	fireEvent.changeText(getByPlaceholderText('votre@email.fr'), "a@a.a");
	fireEvent.changeText(getByPlaceholderText('********'), "aaaaaaaa");
    fireEvent.press(getByText('CONNEXION'));

	await waitFor(() => expect(changeID).toHaveBeenCalledTimes(1));
	expect(changeID).toBeCalledWith("1005");
	expect(changeToken).toHaveBeenCalledTimes(1);
	expect(changeToken).toBeCalledWith("123456789");
	
  });
});