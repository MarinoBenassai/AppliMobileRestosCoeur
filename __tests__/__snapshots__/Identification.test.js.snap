// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Identification should display login button 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 0,
      },
      Object {
        "alignItems": "center",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 200,
        "width": 200,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#656565",
        "marginTop": 15,
        "textAlign": "center",
      }
    }
  >
    Adresse mail
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCompleteType="email"
    autoCorrect={false}
    keyboardType="email-address"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="votre@email.fr"
    rejectResponderTermination={true}
    style={
      Object {
        "borderBottomColor": "#000000",
        "borderBottomWidth": 1,
        "height": 40,
        "marginBottom": 15,
        "width": 200,
      }
    }
    textContentType="emailAddress"
    underlineColorAndroid="transparent"
  />
  <Text
    style={
      Object {
        "color": "#656565",
        "marginTop": 15,
        "textAlign": "center",
      }
    }
  >
    Mot de passe
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="********"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "borderBottomColor": "#000000",
        "borderBottomWidth": 1,
        "height": 40,
        "marginBottom": 15,
        "width": 200,
      }
    }
    textContentType="password"
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessibilityLabel="Bouton de connexion"
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="LoginButton"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#e7007d",
            },
          ]
        }
      >
        CONNEXION
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    title="mdp"
  >
    <Text
      style={
        Object {
          "color": "#656565",
          "marginTop": 15,
          "textAlign": "center",
        }
      }
    >
      Mot de passe oubli√© ?
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#041584",
            },
          ]
        }
      >
        PASSER
      </Text>
    </View>
  </View>
</View>
`;
